src/Databases/PostgreSQL/fetchPostgreSQLTableStructure.cpp

```diff
@@ -135,8 +136,27 @@ static DataTypePtr convertPostgreSQLDataType(String & type, Fn<void()> auto && r
         }
     }
 
+    /// If the type is not mapped so far, assume it's some sort of String
     if (!res)
-        res = std::make_shared<DataTypeString>();
+    {
+        if (type.ends_with(')'))
```

Comment:
`geometry(Point, 4326)` would like a word

Reply:
We need a better way than using parenthesis

Reply:
I'll leave it then to the original implementation that I had where I was checking that the type started with `character`. I thought moving it down here would make it more flexible for some other way of specifying strings with particular sizes, but I see it could bring more trouble than anything.

This is the commit [`050f74a` (#77304)](https://github.com/ClickHouse/ClickHouse/pull/77304/commits/050f74a85fe6c8c1dd331e64d3694d5269edcaae)

Reply:
But this is again not correct -> https://www.postgresql.org/docs/current/datatype-character.html

`character(n), char(n), bpchar(n)` are fixed strings.
`character varying(n), varchar(n)` are variable length with limit (so they should not be matched to CH's FixedString).

Let's hardcode good assumptions instead of using conditionals

Reply:
>character(n), char(n), bpchar(n) are fixed strings.
>Let's hardcode good assumptions instead of using conditionals

Roger, all those 3 can be filtered by checking if the type starts with `character(`.